app/Models/Album.php

```diff
@@ -123,7 +123,8 @@ public function tracks()
     }
 
     public function trackFiles() {
-        return $this->hasManyThrough(TrackFile::class, Track::class, 'album_id', 'track_id');
+        $trackIds = $this->tracks->lists('id');
+        return TrackFile::join('tracks', 'tracks.current_version', '=', 'track_files.version')->whereIn('track_id', $trackIds);
```

Comment:
Changing this to return a query object instead of a relation object is very risky. You should be able to limit the relation to returning only the current version's TrackFiles by chaining a `where` clause onto the `hasManyThrough` method.